File: decompiled/sources/com/hjq/permissions/IPermissionInterceptor.java
package com.hjq.permissions;

import android.app.Activity;
import java.util.ArrayList;
import java.util.List;

public interface IPermissionInterceptor {
    void deniedPermissions(Activity activity, List<String> list, List<String> list2, boolean z, OnPermissionCallback onPermissionCallback);

    void grantedPermissions(Activity activity, List<String> list, List<String> list2, boolean z, OnPermissionCallback onPermissionCallback);

    void requestPermissions(Activity activity, List<String> list, OnPermissionCallback onPermissionCallback);

    /* renamed from: com.hjq.permissions.IPermissionInterceptor$-CC  reason: invalid class name */
    public final /* synthetic */ class CC {
        public static void $default$requestPermissions(IPermissionInterceptor _this, Activity activity, List list, OnPermissionCallback onPermissionCallback) {
            PermissionFragment.beginRequest(activity, new ArrayList(list), _this, onPermissionCallback);
        }

        public static void $default$grantedPermissions(IPermissionInterceptor _this, Activity activity, List list, List list2, boolean z, OnPermissionCallback onPermissionCallback) {
            if (onPermissionCallback != null) {
                onPermissionCallback.onGranted(list2, z);
            }
        }

        public static void $default$deniedPermissions(IPermissionInterceptor _this, Activity activity, List list, List list2, boolean z, OnPermissionCallback onPermissionCallback) {
            if (onPermissionCallback != null) {
                onPermissionCallback.onDenied(list2, z);
            }
        }
    }
}

File: decompiled/sources/com/hjq/permissions/IPermissionInterceptor.kt
package com.hjq.permissions

import android.app.Activity
import java.util.ArrayList

interface IPermissionInterceptor {
    fun deniedPermissions(
        activity: Activity,
        allPermissions: List<String>,
        deniedPermissions: List<String>,
        doNotAskAgain: Boolean,
        callback: OnPermissionCallback?
    ) {
        callback?.onDenied(deniedPermissions, doNotAskAgain)
    }

    fun grantedPermissions(
        activity: Activity,
        allPermissions: List<String>,
        grantedPermissions: List<String>,
        allGranted: Boolean,
        callback: OnPermissionCallback?
    ) {
        callback?.onGranted(grantedPermissions, allGranted)
    }

    fun requestPermissions(
        activity: Activity,
        permissions: List<String>,
        callback: OnPermissionCallback?
    ) {
        PermissionFragment.beginRequest(activity, ArrayList(permissions), this, callback)
    }

    object CC {
        @JvmStatic
        fun `$default$requestPermissions`(
            interceptor: IPermissionInterceptor,
            activity: Activity,
            permissions: List<String>,
            callback: OnPermissionCallback?
        ) {
            PermissionFragment.beginRequest(activity, ArrayList(permissions), interceptor, callback)
        }

        @JvmStatic
        fun `$default$grantedPermissions`(
            interceptor: IPermissionInterceptor,
            activity: Activity,
            allPermissions: List<String>,
            grantedPermissions: List<String>,
            allGranted: Boolean,
            callback: OnPermissionCallback?
        ) {
            callback?.onGranted(grantedPermissions, allGranted)
        }

        @JvmStatic
        fun `$default$deniedPermissions`(
            interceptor: IPermissionInterceptor,
            activity: Activity,
            allPermissions: List<String>,
            deniedPermissions: List<String>,
            doNotAskAgain: Boolean,
            callback: OnPermissionCallback?
        ) {
            callback?.onDenied(deniedPermissions, doNotAskAgain)
        }
    }
}
